crates/holochain/src/core/workflow/app_validation_workflow/tests.rs

```diff
@@ -1,41 +1,156 @@
+use crate::core::queue_consumer::WorkComplete;
 use crate::core::ribosome::guest_callback::validate::ValidateResult;
 use crate::core::ribosome::ZomeCallInvocation;
-use crate::core::workflow::app_validation_workflow::check_app_entry_def;
-use crate::core::SysValidationError;
-use crate::core::ValidationOutcome;
-use crate::sweettest::SweetConductorBatch;
-use crate::sweettest::SweetDnaFile;
-use crate::test_utils::consistency_10s;
-use crate::test_utils::host_fn_caller::*;
-use crate::test_utils::new_invocation;
-use crate::test_utils::new_zome_call;
-use crate::test_utils::wait_for_integration;
+use crate::core::workflow::app_validation_workflow::{
+    app_validation_workflow_inner, check_app_entry_def, put_validation_limbo,
+    AppValidationWorkspace,
+};
+use crate::core::workflow::sys_validation_workflow::validation_query;
+use crate::core::{SysValidationError, ValidationOutcome};
+use crate::sweettest::{
+    SweetConductor, SweetConductorBatch, SweetConductorConfig, SweetDnaFile, SweetLocalRendezvous,
+};
+use crate::test_utils::{
+    consistency_10s, host_fn_caller::*, new_invocation, new_zome_call, wait_for_integration,
+};
+use ::fixt::fixt;
 use arbitrary::Arbitrary;
 use hdk::hdi::test_utils::set_zome_types;
 use hdk::prelude::*;
-use holo_hash::ActionHash;
-use holo_hash::AnyDhtHash;
-use holo_hash::EntryHash;
+use holo_hash::{fixt::AgentPubKeyFixturator, ActionHash, AnyDhtHash, DhtOpHash, EntryHash};
 use holochain_conductor_api::conductor::paths::DataRootPath;
-use holochain_state::prelude::from_blob;
-use holochain_state::prelude::StateQueryResult;
+use holochain_p2p::actor::HolochainP2pRefToDna;
+use holochain_sqlite::error::DatabaseResult;
+use holochain_state::mutations::insert_op;
+use holochain_state::prelude::{from_blob, StateQueryResult};
 use holochain_state::test_utils::test_db_dir;
+use holochain_state::validation_db::ValidationStage;
+use holochain_types::dht_op::{DhtOp, DhtOpHashed};
 use holochain_types::inline_zome::InlineZomeSet;
 use holochain_types::prelude::*;
-use holochain_wasm_test_utils::TestWasm;
-
-use holochain_sqlite::error::DatabaseResult;
-use holochain_wasm_test_utils::TestWasmPair;
-use holochain_wasm_test_utils::TestZomes;
+use holochain_wasm_test_utils::{TestWasm, TestWasmPair, TestZomes};
+use holochain_zome_types::action::Dna;
+use holochain_zome_types::fixt::SignatureFixturator;
+use holochain_zome_types::timestamp::Timestamp;
+use holochain_zome_types::Action;
 use matches::assert_matches;
-use rusqlite::named_params;
-use rusqlite::Transaction;
-use std::convert::TryFrom;
-use std::convert::TryInto;
+use rusqlite::{named_params, Transaction};
+use std::convert::{TryFrom, TryInto};
+use std::sync::Arc;
 use std::time::Duration;
 
+#[cfg(test)]
+#[tokio::test(flavor = "multi_thread")]
+async fn main_loop_app_validation_workflow() {
+    use holochain_sqlite::error::DatabaseError;
+    use rusqlite::params;
+
+    holochain_trace::test_run().ok();
```

Comment:
I prefer unwrap here, if my logging isn't working the way I've configured it, I'd rather know

Reply:
Defo, I too would like to know. I misassumed what `ok()` would do.